### PR TITLE
Installer: add capi variant (aws)

### PR DESCRIPTION
**Pull request by patrickdillon**
Add variant to skip building terraform and opt-in to building capi providers

### CODE REVIEW

File: ci-operator/config/openshift/installer/openshift-installer-master__capi.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      cpu: "3"
+      memory: 5Gi
+tests:
+- always_run: false
+  as: e2e-aws-ovn
+  optional: true
+  steps:
+    cluster_profile: aws
+    env:
+      FEATURE_GATES: ClusterAPIInstall=true
```
Comment (vincepri) [inline ci-operator/config/openshift/installer/openshift-installer-master__capi.yaml:75]:
```suggestion
      FEATURE_GATES: [ 'ClusterAPIInstall=true' ]
```

**Comment by patrickdillon** [comment]
/hold

We need to figure out how we're going to enable building the CAPI dependencies.

File: ci-operator/config/openshift/installer/openshift-installer-master__capi.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+build_root:
+  from_repository: true
+images:
+- build_args:
+  - name: OPENSHIFT_INSTALL_CLUSTER_API
+    value: "true"
+  - name: SKIP_TERRAFORM
+    value: "y"
+  dockerfile_path: images/installer/Dockerfile.ci
+  from: base
```
Comment (r4f4) [inline ci-operator/config/openshift/installer/openshift-installer-master__capi.yaml:23]: I think we're supposed to drop this.

**Review by r4f4 — COMMENTED** [review]
The terraform providers are being built:
```
2023-12-09T12:17:19.853385034Z + make -C terraform all
```
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/46602/rehearse-46602-pull-ci-openshift-installer-master-capi-images/1733457167182204928/artifacts/build-logs/installer-amd64.log

Is this error expected:
```
level=fatal msg=failed to fetch Cluster: failed to generate asset "Cluster": failed to run cluster api system: failed to run local control plane: failed to unpack cluster-api binary: failed to open cluster api zip from mirror: open mirror/cluster-api.zip: file does not exist
Installer exit with code 1
```

**Comment by patrickdillon** [comment]
Pointed this now to `installer-altinfra` image.
Depends on https://github.com/openshift/installer/pull/7823

**Comment by patrickdillon** [comment]
/close
In favor of https://github.com/openshift/release/pull/48135

**Pull request closed without merging**